stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt

```diff
@@ -160,4 +170,96 @@ public class ChannelListView @JvmOverloads constructor(
             this.enabled = enabled
         }
     }
+
+    public interface SwipeListener {
+        /**
+         * Invoked when a swipe is detected.
+         *
+         * @param viewHolder the view holder that is being swiped
+         * @param adapterPosition the internal adapter position of the item being bound
+         * @param x the raw X of the swipe origin
+         * @param y the raw Y of the swipe origin
+         */
+        public fun onSwipeStarted(viewHolder: RecyclerView.ViewHolder, adapterPosition: Int, x: Float, y: Float)
+
+        /**
+         * Invoked after a swipe has been detected, and movement is occurring.
+         *
+         * @param viewHolder the view holder that is being swiped
+         * @param adapterPosition the internal adapter position of the item being bound
+         * @param dX the change from the previous swipe touch event to the current
+         */
+        public fun onSwipeChanged(viewHolder: RecyclerView.ViewHolder, adapterPosition: Int, dX: Float)
+
+        /**
+         * Invoked when a swipe is successfully completed naturally, without cancellation.
+         *
+         * @param viewHolder the view holder that is being swiped
+         * @param adapterPosition the internal adapter position of the item being bound
+         * @param x the raw X of the swipe completion
+         * @param y the raw Y of the swipe completion
+         */
+        public fun onSwipeCompleted(viewHolder: RecyclerView.ViewHolder, adapterPosition: Int, x: Float, y: Float)
+
+        /**
+         * Invoked when a swipe is canceled.
+         *
+         * @param viewHolder the view holder that is being swiped
+         * @param adapterPosition the internal adapter position of the item being bound
+         * @param x the raw X of the event that triggered cancellation
+         * @param y the raw Y of the event that triggered cancellation
+         */
+        public fun onSwipeCanceled(viewHolder: RecyclerView.ViewHolder, adapterPosition: Int, x: Float, y: Float)
+
+        /**
+         * Invoked in order to set the [viewHolder]'s initial state when bound. This supports view holder reuse.
+         * When items are scrolled off-screen and the view holder is reused, it becomes important to
+         * track the swiped state and determine if the view holder should appear as swiped for the item
+         * being bound.
+         *
+         * @param viewHolder the view holder being bound
+         * @param adapterPosition the internal adapter position of the item being bound
+         */
+        public fun onRestoreSwipePosition(viewHolder: RecyclerView.ViewHolder, adapterPosition: Int)
+
+        public companion object {
+
+            public val DEFAULT: SwipeListener = object : SwipeListener {
```

Comment:
I suppose that's true. I was trying to keep things the same as the other listeners. If I change this, should I change the other listener defaults as well? I sort of like the companion route with the other functional interfaces because we can declare them with lambdas 
```
        public companion object {
            public val DEFAULT: ChannelClickListener = ChannelClickListener {}
        }
```
instead of 
```
        public object Default : ChannelClickListener {
            override fun onClick(channel: Channel) {
                // no-op
            }
        }
```

Reply:
Hmm, you're right, those are nicer that way. Let's do the `val` in the companions then 🙂